### PR TITLE
Fix focusing of disabled buttons.

### PR DESCRIPTION
**Pull request by knight-bubble**
[`* `#17885` Focus on relations tab jumps to beginning of tab when work package is selected as relation`](http://community.openproject.org/work_packages/17885)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://hirtie-maxim-fix-17885-relation-tab-jumps.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hirtie-maxim-fix-17885-relation-tab-jumps.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: frontend/tests/unit/tests/ui_components/focus-directive-test.js [hunk 1]
```diff
@@ -61,10 +58,22 @@ describe('focus Directive', function() {
 
   describe('element', function() {
     it('should focus the element', function() {
-      compile();
+      compile(element)(scope);
+      scope.$digest();
+
+      timeout.flush();
 
       // NOTE: $(element).is(':focus') is broken in PhantomJS
```
Comment (0xF013) [inline frontend/tests/unit/tests/ui_components/focus-directive-test.js:66]: @hirtie-maxim could we add a protractor test?

File: frontend/tests/integration/work-package-details-relations-spec.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      });
+
+      it('focuses on disabled button', function() {
+        $('.relation[title="Parent"] button').getAttribute('disabled').then(function(disabled){
+          expect(disabled).to.be.equal('true');
+        });
+      })
+    });
+  });
+});
```
Comment (0xF013) [inline frontend/tests/integration/work-package-details-relations-spec.js]: @hirtie-maxim please add \r

**Comment by floriank** [comment]
Okay, i cannot review this at the moment, as the call to the autocomplete for the work package relations is broken on `dev` (and in this branch, as well).

The request against the autocomplete path happens [here](https://github.com/opf/openproject/blob/dev/frontend/app/work_packages/tabs/add-work-package-relation-directive.js#L40):

``` javascript
// [...]
var params = {
  q: term,
  scope: 'all',
  escape: false,
  id: scope.handler.workPackage.props.id,
  // Project Id is undefined
  'project_id': scope.handler.workPackage.props.projectId
};
return $http({
  method: 'GET',
  url: URI(PathHelper.workPackageJsonAutoCompletePath())
    .search(params)
    .toString()
// [...]
```

This results in a call against `WorkPackages::AutoCompletesController` without a `project_id` - resulting in a 404.

We could reproduce this problem with the current `dev` as well as this branch on my machine and @ulferts machine.

@hirtie-maxim could you have a look on why there is no `project_id` transmitted?

**Comment by floriank** [comment]
I opened a ticket for this problem spearately:

https://community.openproject.org/work_packages/18885

**Comment by floriank** [comment]
@hirtie-maxim Could you rebase against the current `dev`, as #2624 has been merged?

File: frontend/tests/integration/work-package-details-relations-spec.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+var chai = require('chai');
```
Comment (myabc) [inline frontend/tests/integration/work-package-details-relations-spec.js:1]: @floriank Merging #2617 means Hound no longer picks up the right `.jshintrc` settings... We should adjust the `config_file` path in `.hound.yml`.

Comment (floriank) [inline frontend/tests/integration/work-package-details-relations-spec.js:1]: #2628 would do that.

Comment (myabc) [inline frontend/tests/integration/work-package-details-relations-spec.js:1]: @floriank merci

**Comment by knight-bubble** [comment]
@floriank checked on local machine, everything goes fine.

**Comment by floriank** [comment]
Thanks!

**Pull request merged**